src/document_basic.c

```diff
@@ -243,7 +244,7 @@ int Document_LoadSchemaFieldJson(Document *doc, RedisSearchCtx *sctx, QueryError
   size_t nitems = sctx->spec->numFields;
   JSONResultsIterator jsonIter = NULL;
 
-  RedisModuleKey *k = RedisModule_OpenKey(sctx->redisCtx, doc->docKey, REDISMODULE_READ);
+  RedisModuleKey *k = RedisModule_OpenKey(sctx->redisCtx, doc->docKey, REDISMODULE_READ | REDISMODULE_OPEN_KEY_NOEFFECTS);
```

Comment:
Maybe we also want the noexpire flag here?
WDYT?

Reply:
isn't it part of no effect?

Reply:
I don't think so because then we would avoid testing the expiration time, and we would index the expired doc in vain.